packages/vite/src/node/server/searchRoot.ts

```diff
@@ -0,0 +1,49 @@
+import fs from 'fs'
+import { dirname } from 'path'
+import { join } from 'path'
+
+// https://github.com/vitejs/vite/issues/2820#issuecomment-812495079
+const ROOT_FILES = [
+  '.git',
+
+  // https://pnpm.js.org/workspaces/
+  'pnpm-workspace.yaml'
+
+  // https://rushjs.io/pages/advanced/config_files/
+  // 'rush.json',
+
+  // https://nx.dev/latest/react/getting-started/nx-setup
+  // 'workspace.json',
+  // 'nx.json'
+]
+
+// npm: https://docs.npmjs.com/cli/v7/using-npm/workspaces#installing-workspaces
+// yarn: https://classic.yarnpkg.com/en/docs/workspaces/#toc-how-to-use-it
+function hasWorkspacePackageJSON(root: string): boolean {
+  const path = join(root, 'package.json')
+  try {
+    fs.accessSync(path, fs.constants.R_OK)
+  } catch (e) {
```

Comment:
If you want to be fancy and modern, I think you can omit the `(e)` cause we don't use it in the catch body 😸 

```suggestion
  } catch {
```